pkg/hooks/manager.go

```diff
@@ -260,75 +279,152 @@ func (m *hookManager) onDefineDomainCallback(callback *callBackClient, domainSpe
+		}
 
-				conn, err := grpcutil.DialSocketWithTimeout(callback.SocketPath, 1)
-				if err != nil {
-					log.Log.Reason(err).Errorf(dialSockErr, callback.SocketPath)
-					return cloudInitData, err
-				}
-				defer conn.Close()
-
-				client := hooksV1alpha2.NewCallbacksClient(conn)
-				ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-				defer cancel()
-				result, err := client.PreCloudInitIso(ctx, &hooksV1alpha2.PreCloudInitIsoParams{
-					CloudInitData:          cloudInitDataJSON,
-					CloudInitNoCloudSource: cloudInitNoCloudSourceJSON,
-					Vmi:                    vmiJSON,
-				})
-				if err != nil {
-					log.Log.Reason(err).Error("Failed to call PreCloudInitIso")
-					return cloudInitData, err
-				}
+		return [3][]byte{cloudInitDataJSON, cloudInitNoCloudSourceJSON, vmiJSON}, nil
+	}
 
-				err = json.Unmarshal(result.GetCloudInitData(), &resultData)
-				if err != nil {
-					log.Log.Reason(err).Error("Failed to unmarshal CloudInitData result")
-					return cloudInitData, err
-				}
-				if !cloudinit.IsValidCloudInitData(resultData) {
-					// Be backwards compatible for hook sidecars still working on CloudInitNoCloudSource objects instead of CloudInitData
-					var resultNoCloudSourceData *v1.CloudInitNoCloudSource
-					err = json.Unmarshal(result.GetCloudInitNoCloudSource(), &resultNoCloudSourceData)
-					if err != nil {
-						log.Log.Reason(err).Error("Failed to unmarshal CloudInitNoCloudSource result")
-						return cloudInitData, err
-					}
-					resultData = &cloudinit.CloudInitData{
-						DataSource:  cloudInitData.DataSource,
-						UserData:    resultNoCloudSourceData.UserData,
-						NetworkData: resultNoCloudSourceData.NetworkData,
-					}
-				}
-				return resultData, nil
-			} else {
-				panic("Should never happen, version compatibility check is done during Info call")
+	validateResult := func(initData, noCloudSource []byte) (*cloudinit.CloudInitData, error) {
+		var resultData *cloudinit.CloudInitData
+		err := json.Unmarshal(initData, &resultData)
+		if err != nil {
+			log.Log.Reason(err).Error("Failed to unmarshal CloudInitData result")
+			return nil, err
+		}
+
+		if !cloudinit.IsValidCloudInitData(resultData) {
+			// Be backwards compatible for hook sidecars still working on CloudInitNoCloudSource objects instead of CloudInitData
+			var resultNoCloudSourceData *v1.CloudInitNoCloudSource
+			err = json.Unmarshal(noCloudSource, &resultNoCloudSourceData)
+			if err != nil {
+				log.Log.Reason(err).Error("Failed to unmarshal CloudInitNoCloudSource result")
+				return nil, err
 			}
+			resultData = &cloudinit.CloudInitData{
+				DataSource:  cloudInitData.DataSource,
+				UserData:    resultNoCloudSourceData.UserData,
+				NetworkData: resultNoCloudSourceData.NetworkData,
+			}
+		}
+		return resultData, nil
+	}
+
+	callbacks, found := m.CallbacksPerHookPoint[hooksInfo.PreCloudInitIsoHookPointName]
+	if !found {
+		return cloudInitData, nil
+	}
```

Comment:
I am not sure it was necessary to change this from what it was originally here.

```suggestion
	if callbacks, found := m.CallbacksPerHookPoint[hooksInfo.PreCloudInitIsoHookPointName]; found {
```

Reply:
Yes, this is a matter of preference, to return early when possible and reduce identation level. Considering I'm refactoring to add switch plus v1aplha3, I thought it would be fine to do, but not a necessity indeed.